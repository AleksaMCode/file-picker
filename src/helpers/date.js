import { DateTime } from 'luxon'

// TODO: Investiage if we can set language manually
// Getting the browser set language is not a good UX
// Passing as a prop doesn't seem like a nice solution

export function formDateFromNow(date) {
  return DateTime.fromJSDate(new Date(date)).toRelative()
}
