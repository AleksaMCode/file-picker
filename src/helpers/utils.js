export const isSearchParamTruthy = (param) => {
  const urlParams = new URLSearchParams(window.location.search)
  return ['True', 'true', true, '1', 1].includes(urlParams.get(param))
}
