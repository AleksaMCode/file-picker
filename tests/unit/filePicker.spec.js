import { mount, createLocalVue } from '@vue/test-utils'

import { listResources } from '../helpers/mocks'
import { stubs } from '../helpers/stubs'

import FilePicker from '@/components/FilePicker.vue'

const localVue = createLocalVue()

localVue.prototype.$client = {
  files: {
    list: listResources
  }
}

describe('File picker', () => {
  it('renders list of resources', async () => {
    const wrapper = mount(FilePicker, {
      localVue,
      propsData: {
        variation: 'resource'
      },
      stubs
    })

    // Wait twice to give the list of resources enough time to render
    await wrapper.vm.$nextTick()
    await wrapper.vm.$nextTick()

    expect(wrapper.findAll('[filename="ownCloud Manual.pdf"]').length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('emits selected resources', async () => {
    const wrapper = mount(FilePicker, {
      localVue,
      propsData: {
        variation: 'resource'
      },
      stubs
    })

    // Wait twice to give the list of resources enough time to render
    await wrapper.vm.$nextTick()
    await wrapper.vm.$nextTick()

    // For test purpose set only the folder name instead of the whole object
    await wrapper.setData({ selectedResources: 'Documents' })

    // Emit click event instead of calling `trigger()` due to stubbed component
    wrapper.find('.file-picker-btn-select-resources').vm.$emit('click')

    await wrapper.vm.$nextTick()

    // Need to access nested array
    expect(wrapper.emitted().selectResources[0][0]).toContain('Documents')
  })
})
