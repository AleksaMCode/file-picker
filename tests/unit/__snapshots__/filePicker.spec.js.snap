// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File picker renders list of resources 1`] = `
<div class="uk-height-1-1 uk-flex uk-flex-column uk-overflow-hidden">
  <header class="uk-background-primary uk-padding-small uk-flex uk-flex-middle uk-flex-between">
    <oc-breadcrumb-stub items="[object Object]" class="oc-light"></oc-breadcrumb-stub>
    <div>
      <oc-button-stub variation="primary" disabled="true" uk-tooltip="Please select at least one resource. You can select a resource by clicking on its row or via its checkbox." class="file-picker-btn-select-resources">
        Select resources
      </oc-button-stub>
    </div>
  </header>
  <!---->
  <div class="vue-recycle-scroller uk-flex-1 oc-border ready direction-vertical" current-folder="[object Object]">
    <!---->
    <div class="vue-recycle-scroller__item-wrapper" style="min-height: 204px;">
      <div class="vue-recycle-scroller__item-view" style="transform: translateY(0px);">
        <div data-is-visible="true" class="oc-cursor-pointer">
          <oc-grid-stub id="file-row-144227" gutter="small" flex="" class="uk-padding-small oc-border-top">
            <div>
              <oc-checkbox-stub label="Select Documents" hide-label="true" class="file-picker-resource-checkbox uk-margin-small-left"></oc-checkbox-stub>
            </div>
            <div class="uk-flex-inline uk-flex-middle file-picker-resource uk-width-auto oc-file">
              <oc-icon-stub name="folder" size="medium" aria-hidden="true" class="uk-margin-small-right"></oc-icon-stub>
              <div class="uk-width-expand">
                <div filename="Documents" class="file-row-name uk-text-truncate"><span role="button" class="uk-text-bold oc-cursor-pointer oc-file-name uk-padding-remove-left">Documents</span>
                  <!---->
                </div>
                <div class="uk-text-meta">
                  0 B - Last modified
                  11 days ago
                </div>
              </div>
            </div>
          </oc-grid-stub>
        </div>
      </div>
      <div class="vue-recycle-scroller__item-view" style="transform: translateY(68px);">
        <div data-is-visible="true" class="oc-cursor-pointer">
          <oc-grid-stub id="file-row-144228" gutter="small" flex="" class="uk-padding-small oc-border-top">
            <div>
              <oc-checkbox-stub label="Select Photos" hide-label="true" class="file-picker-resource-checkbox uk-margin-small-left"></oc-checkbox-stub>
            </div>
            <div class="uk-flex-inline uk-flex-middle file-picker-resource uk-width-auto oc-file">
              <oc-icon-stub name="folder" size="medium" aria-hidden="true" class="uk-margin-small-right"></oc-icon-stub>
              <div class="uk-width-expand">
                <div filename="Photos" class="file-row-name uk-text-truncate"><span role="button" class="uk-text-bold oc-cursor-pointer oc-file-name uk-padding-remove-left">Photos</span>
                  <!---->
                </div>
                <div class="uk-text-meta">
                  1.3 MB - Last modified
                  11 days ago
                </div>
              </div>
            </div>
          </oc-grid-stub>
        </div>
      </div>
      <div class="vue-recycle-scroller__item-view" style="transform: translateY(136px);">
        <div data-is-visible="true" class="oc-cursor-pointer">
          <oc-grid-stub id="file-row-144242" gutter="small" flex="" class="uk-padding-small oc-border-top">
            <div>
              <oc-checkbox-stub label="Select ownCloud Manual.pdf" hide-label="true" class="file-picker-resource-checkbox uk-margin-small-left"></oc-checkbox-stub>
            </div>
            <div class="uk-flex-inline uk-flex-middle file-picker-resource uk-width-auto">
              <oc-icon-stub name="application-pdf" size="medium" aria-hidden="true" class="uk-margin-small-right"></oc-icon-stub>
              <div class="uk-width-expand">
                <div filename="ownCloud Manual.pdf" class="file-row-name uk-text-truncate"><span role="button" class="uk-text-bold oc-cursor-pointer oc-file-name uk-padding-remove-left">ownCloud Manual</span><span class="uk-text-meta oc-file-extension">.pdf</span></div>
                <div class="uk-text-meta">
                  5.8 MB - Last modified
                  11 days ago
                </div>
              </div>
            </div>
          </oc-grid-stub>
        </div>
      </div>
    </div>
    <!---->
    <div data-v-b329ee4c="" tabindex="-1" class="resize-observer"><object aria-hidden="true" tabindex="-1" type="text/html" data="about:blank"></object></div>
  </div>
</div>
`;
